Q.1. Create two int type variables, apply addition, subtraction, division and multiplications
and store the results in variables. Then print the data in the following format by calling the
variables:

First variable is __ & second variable is __.

Addition: __ + __ = __

Subtraction: __ - __ = __

Multiplication: __ * __ = __

Division: __ / __ = __

In [4]:
a = 5
b = 3
addition = a + b
subtraction = a - b
multiplication = a * b
division = a / b
print(f"First variable is {a} & second variable is {b}.")
print(f"Addition: {a} + {b} = {addition}")
print(f"Subtraction: {a} - {b} = {subtraction}")
print(f"Multiplication: {a} * {b} = {multiplication}")
print(f"Division: {a} / {b} = {division}")

First variable is 5 & second variable is 3.
Addition: 5 + 3 = 8
Subtraction: 5 - 3 = 2
Multiplication: 5 * 3 = 15
Division: 5 / 3 = 1.6666666666666667


Q.2. What is the difference between the following operators:

    (i) ‘/’ & ‘//’


    (ii) ‘**’ & ‘^’


The / operator performs floating-point division in Python. This means that it returns a floating-point number even if the result is a whole number. For example, 5 / 2 would return 2.5.

The // operator performs floor division in Python. This means that it returns the largest integer less than or equal to the result of the division. For example, 5 // 2 would return 2.

The ** operator is used for exponentiation in Python. It raises the first operand to the power of the second operand. For example, 2 ** 3 would return 8.

The ^ operator is used for bitwise XOR in Python. It returns a binary number where each bit is set to 1 if the corresponding bits of the operands are different, and 0 otherwise.

Q.3. List the logical operators.

The logical operators in Python are:

and,or,not


and: Returns True if both operands are true.

or: Returns True if at least one operand is true.

not: Returns the opposite of the operand’s boolean value.


Q.4. Explain right shift operator and left shift operator with examples.

The right shift operator (>>) and left shift operator (<<) are bitwise operators in Python. They shift the bits of the first operand to the right or left by the number of positions specified by the second operand.

The right shift operator (>>) shifts the bits of the first operand to the right by the number of positions specified by the second operand. The bits that are shifted off the right end are discarded, and the empty positions on the left are filled with zeros. For example:

In [6]:
a=60
b=13
a>>2

15

In this example, a is shifted two positions to the right, resulting in c being assigned the value 15.

The left shift operator (<<) shifts the bits of the first operand to the left by the number of positions specified by the second operand. The bits that are shifted off the left end are discarded, and the empty positions on the right are filled with zeros. For example:

In [7]:
a=60
b=13
a<<2

240

In this example, a is shifted two positions to the left, resulting in c being assigned the value 240.

Q.5. Create a list containing int type data of length 15. Then write a code to check if 10 is
present in the list or not.

In [8]:
my_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]

if 10 in my_list:
    print("10 is present in the list.")
else:
    print("10 is not present in the list.")


10 is present in the list.
